x-pack/plugins/cloud_security_posture/server/fleet_integration/fleet_integration.ts

```diff
@@ -50,31 +50,59 @@ export const onPackagePolicyPostCreateCallback = async (
   logger: Logger,
   packagePolicy: PackagePolicy,
   savedObjectsClient: SavedObjectsClientContract
+): Promise<void> => {
+  await UpdateCspRulesAccordingToCspRuleTemplates(packagePolicy, savedObjectsClient, logger);
+};
+
+export const OnPackagePolicyUpgradeCallback = async (
+  logger: Logger,
+  packagePolicy: PackagePolicy,
+  savedObjectsClient: SavedObjectsClientContract
+): Promise<void> => {
+  await UpdateCspRulesAccordingToCspRuleTemplates(packagePolicy, savedObjectsClient, logger);
+};
+
+const UpdateCspRulesAccordingToCspRuleTemplates = async (
+  packagePolicy: PackagePolicy,
+  savedObjectsClient: SavedObjectsClientContract,
+  logger: Logger
 ): Promise<void> => {
   const benchmarkType = getBenchmarkInputType(packagePolicy.inputs);
 
   // Create csp-rules from the generic asset
-  const existingRuleTemplates: SavedObjectsFindResponse<CspRuleTemplate> =
-    await savedObjectsClient.find({
-      type: CSP_RULE_TEMPLATE_SAVED_OBJECT_TYPE,
-      perPage: 10000,
-      filter: getBenchmarkTypeFilter(benchmarkType),
-    });
+  const ruleTemplates: SavedObjectsFindResponse<CspRuleTemplate> = await savedObjectsClient.find({
+    type: CSP_RULE_TEMPLATE_SAVED_OBJECT_TYPE,
+    perPage: 10000,
+    filter: getBenchmarkTypeFilter(benchmarkType),
+  });
 
-  if (existingRuleTemplates.total === 0) {
-    logger.warn(`expected CSP rule templates to exists for type: ${benchmarkType}`);
-    return;
-  }
+  const oldCspRules: SavedObjectsFindResponse<CspRule> = await savedObjectsClient.find({
+    type: CSP_RULE_SAVED_OBJECT_TYPE,
+    perPage: 10000,
+    filter: createCspRuleSearchFilterByPackagePolicy({
```

Comment:
Wondering about this code, should it include the benchmark as well?

```Typescript
createCspRuleSearchFilterByPackagePolicy({
        packagePolicyId: packagePolicy.id,
        policyId: packagePolicy.policy_id,
      }) +
      `AND` +
      getBenchmarkTypeFilter(benchmarkType),
```

I think not, since our current implementation suggests that only one integration can exist in each policy.